pkg/health/server/health.go

```diff
@@ -33,7 +37,9 @@ func (h *getHealthz) getCiliumStatus() (*ciliumModels.StatusResponse, error) {
 // Handle handles GET requests for /healthz .
 func (h *getHealthz) Handle(params GetHealthzParams) middleware.Responder {
 	log.Debug("Handling request for /healthz")
-
+	if _, err := os.Stat(option.Config.SocketPath); os.IsNotExist(err) {
+		return api.Error(GetHealthzFailedCode, fmt.Errorf("%s\nIs the agent running?", option.Config.SocketPath))
+	}
```

Comment:
Shouldn't this be implemented client-side? If the `GET /healthz` request is already being handled then the CLI has successfully connected to the server already.

Reply:
> The cilium health-server relies on cilium.sock to query the cilium-agent daemon for status information to report health. If this file is deleted, the daemon's /healthz endpoint will still return 200

To be honest, I'm not sure whether it should be implemented on the client-side or server-side.

@thorn3r What do you think?

Reply:
I have a question and would like to ask you about it.

> Shouldn't this be implemented client-side? If the GET /healthz request is already being handled then the CLI has successfully connected to the server already.



I didn’t quite understand what you meant. Are you suggesting implementing it on the server-side? Does server-side refer to this?

https://github.com/cilium/cilium/blob/main/daemon/cmd/status.go#L734

@joestringer

Reply:
I am suggesting to implement client-side. If this is implemented server-side, then that means that the client must first connect to the server to find out if this socket exists. But then if this is actually important for the connection to be successful then connecting to the server will also fail. So then there's no point in implementing an improvement on the server side because the client cannot connect in order for the new logic to work.

Reply:
Related, I'm actually confused by your example above. How is the "before:" case successful if the socket is missing? Maybe that command doesn't actually require this socket in order to work?

Reply:
Maybe @thorn3r has more context here on the right solution.

Reply:
I think server-side is the way to go here. The idea is to update /healthz to report an error when the socket is missing so that kubelet probes can detect this and restart the agent.

@joestringer the embedded `cilium` tool in the agent uses the socket to query the health-server and would fail, whereas `cilium-health` just hits the rest endpoint. So as it stand currently, with the socket missing, `cilium-health` would report healthy but `cilium status` would fail.

Reply:
> The idea is to update /healthz to report an error when the socket is missing so that kubelet probes can detect this and restart the agent.

Yes, that's exactly my idea.